common/core/hwclayer.cpp

```diff
@@ -220,11 +220,24 @@ void HwcLayer::Validate() {
     layer_cache_ &= ~kForceClear;
     layer_cache_ &= ~kDisplayFrameRectChanged;
     layer_cache_ &= ~kSourceRectChanged;
+
+    // From observation: In Android, when the source crop doesn't
+    // begin from (0, 0) the surface damage is already translated
+    // to global display co-ordinates
     if (!surface_damage_.empty() &&
-        (display_frame_width_ < DAMAGE_THRESHOLD &&
-         display_frame_height_ < DAMAGE_THRESHOLD)) {
-      layer_cache_ |= kForceClear;
-      current_rendering_damage_ = display_frame_;
+        ((source_crop_.left == 0) && (source_crop_.top == 0))) {
+      if (display_frame_width_ < DAMAGE_THRESHOLD &&
```

Comment:
You should be able to remove DAMAGE_THRESHOLD W/A also here.

Reply:
Ok, will do it.